client/src/Tools/_framework/ToolPanels/MoveContentToFolder.tsx

```diff
@@ -0,0 +1,200 @@
+import React, { useEffect, useState } from "react";
+import {
+  Text,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  Button,
+  SimpleGrid,
+  Heading,
+  IconButton,
+  HStack,
+} from "@chakra-ui/react";
+import axios from "axios";
+import { ArrowBackIcon } from "@chakra-ui/icons";
+import { useFetcher } from "react-router-dom";
+import { ActivityStructure } from "../Paths/ActivityEditor";
+
+interface activeView {
+  folder: null | {
+    name: string;
+    id: number;
+    parentFolderId: number | null;
+  };
+  contents: {
+    isFolder: boolean;
+    name: string;
+    id: number;
+  }[];
+}
+
+export default function MoveContentToFolder({
+  isOpen,
+  onClose,
+  id,
+  currentParentId,
+}) {
+  // Set when the modal opens
+  const [parentId, setParentId] = useState<number | null>(null);
+  const [contentName, setContentName] = useState<string>("");
+  useEffect(() => {
+    if (isOpen) {
+      // TODO: does this async function needed to be awaited for before we end this effect?
+      updateActiveView(currentParentId, true);
+    }
+  }, [isOpen]);
+
+  // Set whenever the user navigates to another folder
+  const [activeView, setActiveView] = useState<activeView>({
+    folder: null,
+    contents: [],
+  });
+  async function updateActiveView(
+    newActiveFolderId: number | null,
+    modalJustOpened: boolean = false,
+  ) {
+    const { data } = await axios.get(
+      `/api/getMyFolderContent/${newActiveFolderId ?? ""}`,
+    );
+
+    const folderName: string = data.folder?.name ?? null;
```

Comment:
The type of `folderName` should be `string | null`.
The type of `parentFolderId` should be `number | null`.